### PR TITLE
add VFS implementation with Aliyun OSS

### PR DESCRIPTION
**Pull request by ihac**
**Do not merge, just for review and criticism**.

Add a new VFS implementation which enables kops to store state with Aliyun OSS when deploying k8s cluster on AliCloud.

``` go
// get Aliyun OSS client
func NewOSSClient(region Region, internal bool, accessKeyId string, accessKeySecret string, secure bool) *Client
```
For now, users have to set three environment variables, `ALIYUN_ACCESS_KEY_ID`, `ALIYUN_ACCESS_KEY_SECRET` and `ALIYUN_REGION`, to generate a OSS client, while `internal` and `secure` are always `true`.

Unit tests would be added by next Monday.

Signed-off-by: Xiao An <hac@zju.edu.cn>

### CODE REVIEW

**Comment by ihac** [comment]
@LilyFaFa @xh4n3

File: util/pkg/vfs/ossfs.go [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+						client: p.client,
+						bucket: p.bucket,
+						key:    d,
+					}
+					paths = append(paths, child)
+				}
+				if comPrefLen != 0 {
+					lastComPref := resp.CommonPrefixes[comPrefLen-1]
+					if strings.Compare(lastComPref, opt.marker) == 1 {
+						opt.marker = resp.CommonPrefixes[comPrefLen-1]
```
Comment (ihac) [inline util/pkg/vfs/ossfs.go]: change to `opt.marker = lastComPref`

**Comment by ihac** [comment]
Free to merge.

**Comment by xh4n3** [comment]
All look great!

**Pull request merged**